diff --git a/configure.ac b/configure.ac
index b80c6033b..4963800ae 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1195,6 +1195,10 @@ AC_ARG_ENABLE(tnef_parse-plugin,
 		[  --disable-tnef_parse-plugin     Do not build tnef_parse plugin],
 		[enable_tnef_parse_plugin=$enableval], [enable_tnef_parse_plugin=auto])
 
+AC_ARG_ENABLE(virustotal-plugin,
+		[  --disable-virustotal-plugin     Do not build virustotal plugin],
+		[enable_virustotal_plugin=$enableval], [enable_virustotal_plugin=auto])
+
 AC_ARG_ENABLE(vcalendar-plugin,
 		[  --disable-vcalendar-plugin      Do not build vcalendar plugin],
 		[enable_vcalendar_plugin=$enableval], [enable_vcalendar_plugin=auto])
@@ -1227,6 +1231,7 @@ dnl S/Mime:		pgpcore libgpgme
 dnl Python:		Python
 dnl RSSyl:		expat libcurl
 dnl SpamReport:		libcurl
+dnl VirusTotal:		libcurl
 dnl vCalendar:		libcurl, libical
 dnl tnef_parse: 	libytnef
 
@@ -2073,6 +2078,32 @@ else
 fi
 
 
+AC_MSG_CHECKING([whether to build virustotal plugin])
+if test x"$enable_virustotal_plugin" != xno; then
+	dependencies_missing=""
+
+	if test x"$HAVE_CURL" = xno; then
+		dependencies_missing="libcurl $dependencies_missing"
+	fi
+
+	if test x"$dependencies_missing" = x; then
+		PLUGINS="$PLUGINS virustotal"
+		AC_MSG_RESULT(yes)
+	elif test x"$enable_virustotal_plugin" = xauto; then
+		AC_MSG_RESULT(no)
+		AC_MSG_WARN("Plugin virustotal will not be built; missing $dependencies_missing")
+		enable_virustotal_plugin=no
+		MISSING_DEPS_PLUGINS="$MISSING_DEPS_PLUGINS virustotal"
+	else
+		AC_MSG_RESULT(no)
+		AC_MSG_ERROR("Plugin virustotal cannot be built; missing $dependencies_missing")
+	fi
+else
+	DISABLED_PLUGINS="$DISABLED_PLUGINS virustotal"
+	AC_MSG_RESULT(no)
+fi
+
+
 AC_MSG_CHECKING([whether to build vcalendar plugin])
 if test x"$enable_vcalendar_plugin" != xno; then
 	dependencies_missing=""
@@ -2139,6 +2170,7 @@ AM_CONDITIONAL(BUILD_SMIME_PLUGIN,		test x"$enable_smime_plugin" != xno)
 AM_CONDITIONAL(BUILD_SPAMASSASSIN_PLUGIN,	test x"$enable_spamassassin_plugin" != xno)
 AM_CONDITIONAL(BUILD_SPAM_REPORT_PLUGIN,	test x"$enable_spam_report_plugin" != xno)
 AM_CONDITIONAL(BUILD_TNEF_PARSE_PLUGIN,		test x"$enable_tnef_parse_plugin" != xno)
+AM_CONDITIONAL(BUILD_VIRUSTOTAL_PLUGIN,		test x"$enable_virustotal_plugin" != xno)
 AM_CONDITIONAL(BUILD_VCALENDAR_PLUGIN,		test x"$enable_vcalendar_plugin" != xno)
 
 
@@ -2216,6 +2248,8 @@ src/plugins/spam_report/Makefile
 src/plugins/spam_report/version.rc
 src/plugins/tnef_parse/Makefile
 src/plugins/tnef_parse/version.rc
+src/plugins/virustotal/Makefile
+src/plugins/virustotal/version.rc
 src/plugins/vcalendar/Makefile
 src/plugins/vcalendar/version.rc
 src/tests/Makefile
diff --git a/src/plugins/Makefile.am b/src/plugins/Makefile.am
index fff6071ed..3d9bbd1d8 100644
--- a/src/plugins/Makefile.am
+++ b/src/plugins/Makefile.am
@@ -36,6 +36,7 @@ SUBDIRS = \
 	spamassassin \
 	spam_report \
 	tnef_parse \
+	virustotal \
 	vcalendar
 
 .PHONY: test
